Update bootstrap.md to contain more information

c1728p9 opened the pull request:
Update the bootstrap page by adding information on linker script changes and adding newer information about the files cmsis.h, cmsis_nvic.h, mbed_rtx.h and targets.json.

c1728p9 commented:
@sg- can you review this?

AnotherButler commented on docs/reference/contributing/target/bootstrap.md:
```diff
@@ -1,10 +1,64 @@
+
+After adding the core files, the next step is to update the linker scripts for `mbed-os`. To do this, follow the steps below.
+
+* Reserve space for RAM vector table. For the example below this is:
```
Query: Do these steps need to happen in order?

c1728p9 replied:
Nope

c1728p9 commented:
> We should add something about the startup file

@bulislaw, are there any modifications that need to be made to the CMSIS-Core startup file? I thought this was compatible with mbed-os.

> I would remove the Finishing section as that's not all you need to do.

Is there more to do to get the target to boot and run or are you just referring to the next porting steps?

> I would remove the point about targets.json as it doesn't belong here (and it's not helping really) we need separate section that would go in details about it.

This is required to get a new target building and booting.

c1728p9 commented:
I rebased this PR and updated it so it has linker script templates for GCC and IAR. I kept the liker script templates simple with just one rom region and one ram region. Adding support for multiple regions and security bits is possible but will be a more involved task that is likely outside the scope of this docs update.

bulislaw commented:
> We should add something about the startup file
> @bulislaw, are there any modifications that need to be made to the CMSIS-Core startup file? I thought this was compatible with mbed-os.

I don't think so, but we don't even mention that it's needed or where it should go, it's required to boot.

> I would remove the Finishing section as that's not all you need to do.
> Is there more to do to get the target to boot and run or are you just referring to the next porting steps?
It's not a stand alone page but part of the porting guide, we will have a section on that, it doesn't need to be duplicated on every page.

> I would remove the point about targets.json as it doesn't belong here (and it's not helping really) we need separate section that would go in details about it.
> This is required to get a new target building and booting.

Again it's not a stand alone document, but part of porting guide.

sg- commented on docs/reference/contributing/target/bootstrap.md:
```diff
@@ -1,10 +1,295 @@
+#### Create and update files
+
+- Extend CMSIS-CORE by adding the file `mbed-os\targets\TARGET_VENDOR\TARGET_MCUNAME\cmsis.h`. This header file includes device-specific headers that include CMSIS-CORE.
+- Add a relocation function in `mbed-os\targets\TARGET_VENDOR\TARGET_MCUNAME\cmsis_nvic.h`. This contains the define `NVIC_NUM_VECTORS`, which is the number of vectors the devices has, and `NVIC_RAM_VECTOR_ADDRESS`, which is the address of the RAM vector table.
```
With vectors defined in the linker script, do we need cmsis_nvic.h anymore?

c1728p9 replied:
This is needed in both places. In the future it would be good to just use a value from the linker script.

sg- commented on docs/reference/contributing/target/bootstrap.md:
```diff
@@ -1,10 +1,295 @@
+
+- Extend CMSIS-CORE by adding the file `mbed-os\targets\TARGET_VENDOR\TARGET_MCUNAME\cmsis.h`. This header file includes device-specific headers that include CMSIS-CORE.
+- Add a relocation function in `mbed-os\targets\TARGET_VENDOR\TARGET_MCUNAME\cmsis_nvic.h`. This contains the define `NVIC_NUM_VECTORS`, which is the number of vectors the devices has, and `NVIC_RAM_VECTOR_ADDRESS`, which is the address of the RAM vector table.
+- Define the initial stack pointer, `INITIAL_SP`, in `mbed_rtx.h`. This file is typically in `mbed-os\targets\TARGET_VENDOR\mbed_rtx.h`.
```
Whats the next step to get rid of mbed_rtx.h, extern a symbol from the .S startup file?

c1728p9 replied:
If a heap + isr stack region is added to the linker script and used from the code then this could be removed.

c1728p9 commented:
@bulislaw Sam is happy with this and I would like to get this merged. I disagree with the changes you suggested but it may be because I don't understand what you are looking for. Could you either post the changes to this PR or approve this PR and make the changes in a followup PR?
